src/explore-education-statistics-common/src/components/form/FormCheckboxGroup.tsx

```diff
@@ -42,118 +43,111 @@ interface BaseFormCheckboxGroupProps {
-          'govuk-checkboxes--small': small,
-        })}
-        ref={this.ref}
-      >
-        {options.length > 1 && selectAll && (
-          <ButtonText
-            id={`${id}-all`}
-            onClick={this.handleAllChange}
-            className={styles.selectAll}
-            underline={false}
-          >
-            {`${this.isAllChecked() ? 'Unselect' : 'Select'} all ${
-              options.length
-            } options`}
-          </ButtonText>
-        )}
+  }, [options, value]);
+
+  const handleAllChange: MouseEventHandler<HTMLButtonElement> = useCallback(
+    event => {
+      if (onAllChange) {
+        onAllChange(event, isAllChecked);
+      }
+    },
+    [isAllChecked, onAllChange],
+  );
 
-        {orderBy(options, order, orderDirection).map(option => (
-          <FormCheckbox
-            disabled={disabled}
-            {...option}
-            id={option.id ? option.id : `${id}-${kebabCase(option.value)}`}
-            name={name}
-            key={option.value}
-            checked={value.indexOf(option.value) > -1}
-            onChange={this.handleChange}
-          />
-        ))}
+  return (
+    <div
+      className={classNames('govuk-checkboxes', {
+        'govuk-checkboxes--small': small,
+      })}
+      ref={ref}
+    >
+      {options.length > 1 && selectAll && (
+        <ButtonText
+          id={`${id}-all`}
+          onClick={handleAllChange}
+          className={styles.selectAll}
+          underline={false}
+        >
+          {`${isAllChecked ? 'Unselect' : 'Select'} all ${
+            options.length
+          } options`}
+        </ButtonText>
+      )}
+
+      {orderBy(options, order, orderDirection).map(option => (
+        <FormCheckbox
+          disabled={disabled}
+          {...option}
+          id={
+            option.id ? option.id : `${id}-${option.value.replace(/\s/g, '-')}`
+          }
+          name={name}
+          key={option.value}
+          checked={value.indexOf(option.value) > -1}
+          onBlur={onBlur}
+          onChange={onChange}
+        />
+      ))}
+
+      {options.length === 0 && <p>No options available.</p>}
+    </div>
+  );
+};
 
-        {options.length === 0 && <p>No options available.</p>}
-      </div>
-    );
-  }
-}
+export type FormCheckboxGroupProps = BaseFormCheckboxGroupProps &
+  OmitStrict<FormFieldsetProps, 'onBlur' | 'onFocus'> & {
+    onFieldsetBlur?: FocusEventHandler<HTMLFieldSetElement>;
+    onFieldsetFocus?: FocusEventHandler<HTMLFieldSetElement>;
+  };
 
-const FormCheckboxGroup = (props: FormCheckboxGroupProps) => {
+const FormCheckboxGroup = ({
+  onFieldsetBlur,
+  onFieldsetFocus,
+  ...props
+}: FormCheckboxGroupProps) => {
   return (
-    <FormFieldset {...props}>
+    <FormFieldset {...props} onBlur={onFieldsetBlur} onFocus={onFieldsetFocus}>
       <BaseFormCheckboxGroup {...props} />
     </FormFieldset>
   );
 };
 
-export default FormCheckboxGroup;
+export default memo(FormCheckboxGroup);
```

Comment:
This is to prevent re-rendering on `FormCheckboxGroup` when its props don't change. Normally React defaults to automatically re-rendering everything in a VDOM subtree whenever it needs to re-render. This isn't usually a problem for read-only pages.

In situations where we will potentially re-render many elements in the UI in response to user input e.g. in forms. This can be a real performance hog. Table tool's checkbox groups are an example of this as there can be potentially 100+ checkboxes for a form. Each of these would re-render whenever the user clicks on any of them.

When re-rendering this many components, the jank can be very noticeable and can lead to clicks seemingly being missed as the UI can't respond fast enough. `memo` thankfully solves this.

If you're thinking about using this in the future, you need to make sure that props are shallowly equal (reference equality for functions and objects). This requires you to use of `useCallback` (for wrapping event handlers/callbacks) and `useMemo` (for objects or other calculated values) in the parent component.